nbqa/replace_source.py

```diff
@@ -91,7 +91,12 @@ def _reinstate_magics(
         New source that can be saved into Jupyter Notebook.
     """
     for magic_substitution in temporary_lines:
-        source = source.replace(magic_substitution.replacement, magic_substitution.src)
+        source = re.sub(
+            f"{re.escape(magic_substitution.replacement)}\n*",
```

Comment:
instead of subbing any number of newlines, let's try to preserve the original number